torchtune/utils/precision.py

```diff
@@ -161,18 +171,20 @@ def set_default_dtype(dtype: torch.dtype) -> Generator[None, None, None]:
         torch.set_default_dtype(old_dtype)
 
 
-def validate_expected_param_dtype(model: torch.nn.Module, dtype: torch.dtype) -> None:
+def validate_expected_param_dtype(
+    named_params: Iterable[Tuple[str, nn.Parameter]], dtype: torch.dtype
+) -> None:
     """
-    Validates that all parameters in the model have the expected dtype.
+    Validates that all input parameters have the expected dtype.
 
     Args:
-        model (torch.nn.Module): Model to validate.
+        named_params (Iterable[Tuple[str, nn.Parameter]]): Iterable of named parameters.
```

Comment:
Any particular reason for the change from model to its named parameters? If not, I personally slightly preferred the model API